arangod/Agency/AgencyComm.cpp

```diff
@@ -177,13 +177,54 @@ AgencyOperation::AgencyOperation(std::string const& key,
   _opType.value = opType;
 }
 
+AgencyOperation::AgencyOperation(std::string const& key, AgencyValueOperationType opType,
+                                 std::shared_ptr<velocypack::Builder> value)
+    : AgencyOperation(std::move(key), opType, value->slice()) {
```

Comment:
I don't think that `key` can be moved here, as it is `const`.

Reply:
I also think this calls the base constructor, which will do a full copy of the VPack slice in `value` into a newly allocated shared_ptr `_holder`, only to overwrite `_holder` later. So `_value` may point into the old `_holder` and it could be invalid.

Reply:
Right, thanks, the move is useless. I think the rest is correct though. This delegates to the constructor `AgencyOperation(std::string const&, AgencyValueOperationType, Slice)`, which does not copy the slice. Only the constructors getting some `T` which is then passed to `VPackValue` do that. I've added assertions to make that clear.